Fix/add new configurations for Baytrail/Cherrytrail tests

plbossart opened the pull request:
The previous PR was merged too quickly, we still had a bunch of issues.

With this PR, along with https://github.com/thesofproject/linux/pull/2021 and https://github.com/alsa-project/alsa-ucm-conf/pull/20, all existing files are handled, but I still get errors with the SOF.conf. It seems the inclusion of other files isn't detected, all the internal if/else cases should already be covered.

````
./ucm.py --level 0 all ../../../alsa-ucm-conf/ucm2
./ucm.py --level 0 configs ../../../alsa-ucm-conf/ucm2 configs 
ERR: SOF/SOF.conf: Verb=HiFi Device=Headphones ConflictingDevices device 'Speaker' is not valid
ERR: SOF/SOF.conf: Verb=HiFi Device=Headphones ConflictingDevices device 'Speaker' is not valid
ERR: SOF/SOF.conf: Verb=HiFi Device=Headphones ConflictingDevices device 'Speaker' is not valid
ERR: SOF/SOF.conf: Verb=HiFi device name "HSMic" unknown (see the specification)
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5650'.'True'.'SectionVerb'.'If'.'Controls'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'spk'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'mono'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5651'.'True'.'If'.'0'.'False'.'If'.'dmic1'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'in1'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht-rt5640'.'True'.'If'.'0'.'False'.'If'.'in3'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5651'.'True'.'If'.'0'.'False'.'If'.'headphones'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'in2'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5651'.'True'.'If'.'0'.'False'.'If'.'in12'.'Condition' - True block not executed
````

@perexg I probably need your help to make this error-free....

plbossart committed:
configs: bytcr-rt5651: add SOF mode in dummy.json

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
configs: bytcht-es8316: add SOF mode in dummy.json

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
configs: bytcht-cx2072x: add SOF mode in dummy.json

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
configs: bytcht-nau8824: add SOF mode in dummy.json

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
configs: cht-bsw-rt5672: add SOF mode in dummy.json

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
configs: chtrt5645: add SOF mode in dummy.json

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
config: chtrt5645: add Lenovo-MIIX-320-sof.txt

Somehow I wasn't able to use the .json solution, so just took the
existing alsa-info file and rename one 	control as
    name 'media0_in GainSOF 0 Switch'

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

plbossart committed:
configs: SOF: add dummy.json

Add basic dummy.json to check if the basic inclusion mechanism works.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

perexg commented:
1.  I fixed the "HSMic" error (wrong chtrt5650 config - it was not checked by ucm-validator somehow before). Please, rebase to master branch.

2. The SOF dummy json config does not define some configs. For example, included file <chtrt5650/HiFi.conf> from SOF/HiFi.conf checks for control 'media0_in Gain 0 Switch' . This control is not defined in python/ucm-validator/configs/SOF/dummy.json . You need to trace all included files.

plbossart commented:
> 1. I fixed the "HSMic" error (wrong chtrt5650 config - it was not checked by ucm-validator somehow before). Please, rebase to master branch.

Thanks, done

> 2. The SOF dummy json config does not define some configs. For example, included file <chtrt5650/HiFi.conf> from SOF/HiFi.conf checks for control 'media0_in Gain 0 Switch' . This control is not defined in python/ucm-validator/configs/SOF/dummy.json . You need to trace all included files.

there are 3 issues here:
a) chtrt5650/chtrt5650.conf did not use Syntax 2, fixed now on my branch.
b) there are issues unrelated to my work
````
ERR: HDA-Intel/HDAudio-DualCodecs.conf@HiFi-dual.conf: 'SectionDevice'.'Line2'.'If'.'0'.'Condition' - True block not executed
ERR: HDA-Intel/HDAudio-Lenovo-DualCodecs.conf@HiFi-dual.conf: 'SectionDevice'.'Line2'.'If'.'0'.'Condition' - True block not executed
ERR: HDA-Intel/HDAudio-Gigabyte-ALC1220DualCodecs.conf@HiFi-dual.conf: 'SectionDevice'.'Line2'.'If'.'0'.'Condition' - True block not executed
````
c) when including a file, do we need to provide ALL the configs previously added for each of those drivers. Is this really necessary? If I am including a file that's been completely checked already, why would I revalidate it again by duplicating everything? I think there should be something at the UCM validator level to detect the low-level configurations are tested already. Also I don't get the issues on ConflictingDevices, this is also not something I changed.

````
./ucm.py --level 0 all ../../../alsa-ucm-conf/ucm2
./ucm.py --level 0 configs ../../../alsa-ucm-conf/ucm2 configs 
ERR: SOF/SOF.conf: Verb=HiFi Device=Headphones ConflictingDevices device 'Speaker' is not valid
ERR:   AlsaJsonSoundcard(id='rt5640', driver='SOF', name='sof-bytcht rt5640', longname='sof-bytcht rt5640')
ERR: SOF/SOF.conf: Verb=HiFi Device=Headphones ConflictingDevices device 'Speaker' is not valid
ERR:   AlsaJsonSoundcard(id='rt5651', driver='SOF', name='sof-bytcht rt5651', longname='sof-bytcht rt5651')
ERR: SOF/SOF.conf: Verb=HiFi Device=Headphones ConflictingDevices device 'Speaker' is not valid
ERR:   AlsaJsonSoundcard(id='es8316', driver='SOF', name='sof-bytcht es8316', longname='sof-bytcht es8316')
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5650'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5645'.'True'.'SectionVerb'.'If'.'Controls'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'spk'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'mono'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5651'.'True'.'If'.'0'.'False'.'If'.'dmic1'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'in1'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht-rt5640'.'True'.'If'.'0'.'False'.'If'.'in3'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5651'.'True'.'If'.'0'.'False'.'If'.'headphones'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_es8316'.'True'.'If'.'0'.'False'.'If'.'in2'.'Condition' - True block not executed
ERR: SOF/SOF.conf@HiFi.conf: 'If'.'bytcht_rt5651'.'True'.'If'.'0'.'False'.'If'.'in12'.'Condition' - True block not executed
ERR: HDA-Intel/HDAudio-DualCodecs.conf@HiFi-dual.conf: 'SectionDevice'.'Line2'.'If'.'0'.'Condition' - True block not executed
ERR: HDA-Intel/HDAudio-Lenovo-DualCodecs.conf@HiFi-dual.conf: 'SectionDevice'.'Line2'.'If'.'0'.'Condition' - True block not executed
ERR: HDA-Intel/HDAudio-Gigabyte-ALC1220DualCodecs.conf@HiFi-dual.conf: 'SectionDevice'.'Line2'.'If'.'0'.'Condition' - True block not executed
ERR: chtrt5650/chtrt5650.conf@HiFi.conf: 'SectionVerb'.'If'.'Controls'.'Condition' - True block not executed
ERR: total errors: 18
````

perexg commented:
I reworked the ucm-validator. The new if errors suppression can control the duplicate configuration paths for SOF. I merged your proposed changes with some cleanups. The alsa-tests and alsa-ucm-conf repo should be in sync now with the SOF configs.
